waiter/src/waiter/util/http_utils.clj

```diff
@@ -102,9 +102,9 @@
       (.clear (.getContentDecoderFactories client)))
     (.setCookieStore client (HttpCookieStore$Empty.))
     (.setDefaultRequestContentType client nil)
-    (when user-agent
-      (let [new-user-agent-field (HttpField. HttpHeader/USER_AGENT (str user-agent))]
-        (.setUserAgentField client new-user-agent-field)))
+    (.setUserAgentField client
+                        (when-not (str/blank? user-agent)
+                          (HttpField. HttpHeader/USER_AGENT (str user-agent))))
```

Comment:
This is the main fix where we are explicitly setting the user-agent field to null instead of letting it be the default.
https://github.com/eclipse/jetty.project/blob/fa54c749466a1a2b032ae284c73ab559493c07d6/jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java#L138